include/envoy/common/crypto/crypto.h

```diff
@@ -0,0 +1,31 @@
+#pragma once
+
+#include <cstdint>
+#include <functional>
+#include <iostream>
+#include <memory>
+
+#include "envoy/common/pure.h"
+
+namespace Envoy {
+namespace Common {
+namespace Crypto {
+
+class CryptoObject {
+public:
+  virtual ~CryptoObject() = default;
+};
+
+using CryptoObjectUniquePtr = std::unique_ptr<CryptoObject>;
+
+namespace Access {
+
+template <class T> T* getTyped(CryptoObjectUniquePtr* cryptoPtr) {
+  return static_cast<T*>(cryptoPtr->get());
```

Comment:
dynamic_cast

Reply:
dynamic_cast fails CI

Reply:
that is because you're taking `void*` below instead of `CryptoObject*`, dynamic_cast from void* always results nullptr.